Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5410,6 +5410,20 @@ void IFileContentManagerHost.ReportContent(FileArtifact artifact, in FileMateria
             }
         }
 
+        [SuppressMessage("Microsoft.Design", "CA1033:InterfaceMethodsShouldBeCallableByChildTypes")]
+        void IFileContentManagerHost.ReportDynamicContent(FileArtifact artifact, PipOutputOrigin origin)
+        {
+            Contract.Requires(artifact.IsValid);
+            Contract.Requires(artifact.IsOutputFile);
+
+            if (origin != PipOutputOrigin.NotMaterialized)
+            {
+                State.FileSystemView.ReportRealFileSystemExistence(artifact.Path, PathExistence.ExistsAsFile);
+            }
+
+            State.FileSystemView.ReportOutputFileSystemExistence(artifact.Path, PathExistence.ExistsAsFile);

```

Comment:
>PathExistence.ExistsAsFile [](start = 80, length = 26)

I'm not sure about "exists as a file" here. We can step into this method with origin == 'PipOrigin.NotMaterialized', and we'll mark the file as existent.

Reply:
removed the method completely

---
In reply to: [286725574](https://github.com/microsoft/BuildXL/pull/361#discussion_r286725574) [](ancestors = 286725574)